pyproject.toml

```
@@ -1,7 +1,6 @@
 [build-system]
 requires = [
-     # pin NumPy version used in the build
-     "numpy>=1.20.1, <2",
+     "numpy>=2.0.0",
```

```suggestion
     "numpy>=1.23",
```

Is there a reason to be stricter here? I'm not actually sure anything in matminer needs to be built with numpy, but imagine this is legacy and its effects are untested

---

> Is there a reason to be stricter here?

Yes we have a build with numpy 2 as ["NumPy’s ABI is forward but not backward compatible"](https://numpy.org/devdocs/dev/depending_on_numpy.html#build-time-dependency):
> When you build wheels for your package using a NumPy 1.xx version at build time, those will not work with NumPy 2.0.
> When you build wheels for your package using a NumPy 2.x version at build time, those will work with NumPy 1.xx.

---

> I'm not actually sure anything in matminer needs to be built with numpy, but imagine this is legacy and its effects are untested

I would have a look later. I'm not sure familiar with the code base, but it seems it's a pure-python package? i.e. no c extersion or cython module and so on?